### PR TITLE
goflow: init at 3.4.3

### PR DESCRIPTION
**Pull request by heph2**
###### Motivation for this change
This application is a NetFlow/IPFIX/sFlow collector in Go.
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/goflow/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  doCheck = false;
+  vendorSha256 = "sha256-fOlfVI8v7KqNSRhAPlZBSHKfZRlCbCgjnMV/6bsqDhg=";
+
+  meta = with lib; {
+    description = "This application is a NetFlow/IPFIX/sFlow collector in Go.";
+    homepage = "https://github.com/cloudflare/goflow";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ heph2 ];
+    platforms = platforms.linux;
```
Comment (papojari) [inline pkgs/tools/networking/goflow/default.nix:25]: Why only Linux? Go can be compile to other OSs too. Or is the program only written for Linux? I couldn't find anything about that in their readme. I suggest:
```suggestion
    platforms = platforms.all;
```

File: pkgs/tools/networking/goflow/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Upxeik2Db7XjPCooCjvBINiAw6Nq+i7oMo9nbG8ecDs=";
+  };
+
+  doCheck = false;
+  vendorSha256 = "sha256-fOlfVI8v7KqNSRhAPlZBSHKfZRlCbCgjnMV/6bsqDhg=";
+
+  meta = with lib; {
+    description = "This application is a NetFlow/IPFIX/sFlow collector in Go.";
```
Comment (papojari) [inline pkgs/tools/networking/goflow/default.nix:21]:
```suggestion
    description = "A NetFlow/IPFIX/sFlow collector in Go";
```
`This application` is kind of unecessary even if it's in the project's readme. Also [no periods at the end](https://nixos.org/manual/nixpkgs/stable/#var-meta-description).

**Review by auroraanna — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/goflow/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/tools/networking/goflow/default.nix:25]: In which case platforms could be dropped entirely.

File: pkgs/tools/networking/goflow/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  version = "3.4.2";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Upxeik2Db7XjPCooCjvBINiAw6Nq+i7oMo9nbG8ecDs=";
+  };
+
+  doCheck = false;
```
Comment (mweinelt) [inline pkgs/tools/networking/goflow/default.nix:17]: What's the reason you disabled the tests?

File: pkgs/tools/networking/goflow/default.nix (hunk 1 above)
Comment (papojari) [reply pkgs/tools/networking/goflow/default.nix:25]: https://nixos.org/manual/nixpkgs/stable/#var-meta-platforms says that there won't be prebuilt binaries then though.

File: pkgs/tools/networking/goflow/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
+}:
+
+buildGoModule rec {
+  pname = "goflow";
+  version = "3.4.2";
```
Comment (mweinelt) [inline pkgs/tools/networking/goflow/default.nix:8]: The latest version is 3.4.3, see the Git tag.

Comment (heph2) [reply pkgs/tools/networking/goflow/default.nix]: yep. Just changed the version :D

Comment (mweinelt) [reply pkgs/tools/networking/goflow/default.nix]: Squash the changes and we're good to go.

Comment (heph2) [reply pkgs/tools/networking/goflow/default.nix]: squashed and edited the PR :D

**Comment by papojari** [comment]
Result of `nixpkgs-review pr 162746` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goflow</li>
  </ul>
</details>

**Pull request merged**